core/lib/src/config/custom_values.rs

```diff
@@ -42,6 +43,53 @@ impl fmt::Display for SecretKey {
     }
 }
 
+#[derive(Copy, Clone, Debug)]
```

Comment:
This needs a docstring. How about:

```suggestion
/// The configured port to serve a Rocket application on.
#[derive(Copy, Clone, Debug)]
```

We should probably implement `From<u16>` for `Port` with an implementation of `Port::Provided(value)`.